pkg/workload/schemachange/optype.go

```diff
@@ -0,0 +1,209 @@
+	if len(opFuncs) != numOpTypes {
+		panic(errors.AssertionFailedf(
+			"len(opFuncs) and numOpTypes don't match but a missing operation wasn't found. Did the definition of numOpTypes change?",
+		))
+	}
+}
+
+const (
+	addColumn               opType = iota // ALTER TABLE <table> ADD [COLUMN] <column> <type>
+	addConstraint                         // ALTER TABLE <table> ADD CONSTRAINT <constraint> <def>
+	addForeignKeyConstraint               // ALTER TABLE <table> ADD CONSTRAINT <constraint> FOREIGN KEY (<column>) REFERENCES <table> (<column>)
+	addRegion                             // ALTER DATABASE <db> ADD REGION <region>
+	addUniqueConstraint                   // ALTER TABLE <table> ADD CONSTRAINT <constraint> UNIQUE (<column>)
+
+	alterTableLocality // ALTER TABLE <table> LOCALITY <locality>
+
+	createIndex    // CREATE INDEX <index> ON <table> <def>
+	createSequence // CREATE SEQUENCE <sequence> <def>
+	createTable    // CREATE TABLE <table> <def>
+	createTableAs  // CREATE TABLE <table> AS <def>
+	createView     // CREATE VIEW <view> AS <def>
+	createEnum     // CREATE TYPE <type> ENUM AS <def>
+	createSchema   // CREATE SCHEMA <schema>
+
+	dropColumn        // ALTER TABLE <table> DROP COLUMN <column>
+	dropColumnDefault // ALTER TABLE <table> ALTER [COLUMN] <column> DROP DEFAULT
+	dropColumnNotNull // ALTER TABLE <table> ALTER [COLUMN] <column> DROP NOT NULL
+	dropColumnStored  // ALTER TABLE <table> ALTER [COLUMN] <column> DROP STORED
+	dropConstraint    // ALTER TABLE <table> DROP CONSTRAINT <constraint>
+	dropIndex         // DROP INDEX <index>@<table>
+	dropSequence      // DROP SEQUENCE <sequence>
+	dropTable         // DROP TABLE <table>
+	dropEnumValue     // ALTER TYPE <type> DROP VALUE <value>
+	dropView          // DROP VIEW <view>
+	dropSchema        // DROP SCHEMA <schema>
+
+	primaryRegion //  ALTER DATABASE <db> PRIMARY REGION <region>
+
+	renameColumn   // ALTER TABLE <table> RENAME [COLUMN] <column> TO <column>
+	renameIndex    // ALTER TABLE <table> RENAME CONSTRAINT <constraint> TO <constraint>
+	renameSequence // ALTER SEQUENCE <sequence> RENAME TO <sequence>
+	renameTable    // ALTER TABLE <table> RENAME TO <table>
+	renameView     // ALTER VIEW <view> RENAME TO <view>
+
+	setColumnDefault // ALTER TABLE <table> ALTER [COLUMN] <column> SET DEFAULT <expr>
+	setColumnNotNull // ALTER TABLE <table> ALTER [COLUMN] <column> SET NOT NULL
+	setColumnType    // ALTER TABLE <table> ALTER [COLUMN] <column> [SET DATA] TYPE <type>
+
+	survive // ALTER DATABASE <db> SURVIVE <failure_mode>
+
+	insertRow // INSERT INTO <table> (<cols>) VALUES (<values>)
+
+	selectStmt // SELECT..
+
+	validate // validate all table descriptors
+
+	// numOpTypes contains the total number of opType entries and is used to
+	// perform runtime assertions about various structures that aid in operation
+	// generation.
+	numOpTypes int = iota
+)
+
+var opFuncs = []func(*operationGenerator, context.Context, pgx.Tx) (*opStmt, error){
```

Comment:
nit: can you help me understand Go syntax here? It's a slice of functions but the initialization seems to be key-value pairs. The current code is indeed a map
```
var opFuncs = map[opType]func(*operationGenerator, context.Context, pgx.Tx) (*opStmt, error){
...
}
```